specs/eventing/overview.md

```diff
@@ -1,97 +1,133 @@
+The Knative Eventing API provides primitives for two common event-processing
+patterns:
+
+- Topology-based event routing ([`messaging.knative.dev`](#messaging))
+
+- Content-based event routing ([`eventing.knative.dev`](#eventing))
+
+Knative Eventing does not directly specify mechanisms for other event-processing
+models, including multi-stage workflows, correlated request-reply, and
+sequential (windowed) event processing; these models could be built using the
+primitives provided by Knative, or Knative could deliver events to an external
+system that implements these models.
+
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
+
+![Overview of objects](images/eventing-overview.svg)
+<!-- Generated from the following UML using PlantUML:
+@startuml
+' Remove shadows
+skinparam shadowing false
+' Set ordering
+left to right direction
+
+usecase Addressable <<Knative Service>>
+usecase Addressable AS A2 <<Kubernetes Service>>
+
+package "eventing.knative.dev" {
+agent Broker
+agent "Trigger" as T1
+agent "Trigger" as T2
+}
+
+Broker <-- T1 : broker
+Broker <-- T2 : broker
+Addressable <-u- T1 : subscriber
+
+package "messaging.knative.dev" {
+agent Channel
+agent "Subscription" as S1
+agent "Subscription" as S2
+}
+
+Channel <-- S1 :channel
+Channel <-- S2 :channel
+Addressable <-u- S2 :subscriber
+A2 <-u- S2 : reply
+@enduml
+-->
+
+## Interface Contracts
+
+In addition to the concrete types described below in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the other resources must support. The following
+interface contracts define resource fragments and partial schemas (required fields
+on an arbitrary API object) which form a basis for Knative Eventing.
+
+### Addressable
+
+**Addressable** resources expose a resource address (HTTP URL) in their `status`
+object. The URL is used as a destination for delivery of events to the resource;
+the exposed URL must implement the [data plane contract](data-plane.md) for
+receiving events.
+
+[**Broker**](#broker) and [**Channel**](#channel) both implement **Addressable**
+to receive events from other components.
+
+### Destination
+
+**Destination** is an interface (resource fragment) which is used consistently
+through Knative Eventing to reference a event delivery destination. A
+Destination eventually resolves the supplied information to a URL, and may be a
+simple URL or relative to an **Addressable** object reference; it also supports
+a Kubernetes Service object reference (as a special case). An absolute URL in a
+Destination may be used to reference cluster-external resources such as a
+virtual machine or SaaS service.
```

Comment:
Can we show exactly what the resource definitions look like and make them normative? This applies to all resources define here. Having just textual descriptions leave too much to the imagination.

Reply:
https://github.com/knative/specs/blob/99bb2e0cb0e33f36dfd3a4475ebf08668fbca2f3/specs/eventing/control-plane.md#addressable-resolution,  https://github.com/knative/specs/blob/99bb2e0cb0e33f36dfd3a4475ebf08668fbca2f3/specs/eventing/control-plane.md#addressable-v1, and https://github.com/knative/specs/blob/99bb2e0cb0e33f36dfd3a4475ebf08668fbca2f3/specs/eventing/control-plane.md#duckv1destination should bring you joy, then.